crowbar_framework/app/models/service_object.rb

```diff
@@ -1188,6 +1202,9 @@ def apply_role(role, inst, in_queue, bootstrap = false)
               next
             end
 
+            # skip adding nodes to the batch unless they are really in the list to be deployed
+            next unless new_elements[role_name].include?(node_name)
```

Comment:
Not a clean cherry-pick :-)

Reply:
weeeeird, thats what I get for doing things quick without double checking :(

Reply:
fixed!